habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java

```diff
@@ -147,6 +193,24 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             log.info("Checking for updates to poetry-monorepo-dependency-plugin...");
             poetryHelper.installPoetryPlugin("poetry-monorepo-dependency-plugin");
         }
+
+        String username = null;
+        String password = null;
+
+        if (StringUtils.isNotEmpty(this.pypiRepoId)) {
+            username = HabushuUtil.findUsernameForServer(this.settings, this.pypiRepoId);
+            password = HabushuUtil.findPasswordForServer(this.settings, this.pypiRepoId, this.decryptPassword);
```

Comment:
S: I would move these two methods to `AbstractHabushuMojo` since they are highly specific to Maven execution, and just pass the credentials into the constructor of this object.

Reply:
This also allows you to remove `pypiRepoId`, `decryptPassword`, and, `settings` from the constructor so it ends up simplifying this class.

Reply:
If I understand your suggestion, my latest push should have addressed this. Those methods are now in AbstractHabushuMojo and are called by ValidatePyEndAndPoetryMojo, which then passes the values down to PyEnvAndPoetrySetup